### ISSUE TITLE
Document more clearly how to do only ipv4/ipv6

### ISSUE THREAD

**Issue by flopacero** [algesten/ureq#416]
Currently `ureq` uses by default `std::io::ToSocketAddrs` to resolve addresses. Agent tries all the returned addresses until it gets response. 

With a local server listening on incoming requests on `127.0.0.1` it gets a bit confusing since you get response in like 2 seconds. It happens because `ToSocketAddrs` first resolves `localhost` to ipv6 `[::1]` and only then to `127.0.0.1`. So `ureq` waits for the first timeout on `[::1]` and only then goes to `127.0.0.1`.

Currently the only option is to use a custom resolver based on `ToSocketAddrs` and to filter out all the ipv6 addresses. Still it would be great if there were some docs improvements docs so this case would be less confusing. And probably add an option to `AgentBuilder` to resolve addresses only to ipv4 or ipv6.

I would be happy to help with any of it if you decide on any or all of the options.

**Comment by algesten** [comment]
The state of controlling DNS resolving in rust `std` is a bit sad. I'm in two minds adding these kinds of functions. On some level, it's application logic to compensate for your system's behavior.

Just a random thought, why not bind the service to `[::1]:<port>` (since I believe that binds 127.0.0.1 as well)?

**Comment by flopacero** [comment]
Agree, it might not be exactly the responsibility of this library to provide all this utility functions. But maybe some note in docs may give some clue on the situation in case anyone face it. At least this case doesn't seem to be super odd.

And yes, binding ipv6 is also a reasonable solution for my case. It was just a bit surprising and I found no hints while resolving it. So I decided a GitHub issue may help someone else as well.

Feel free to close it if you think no action should be undertaken here.
